engine/src/witness/sol/program_swaps_witnessing.rs

```diff
@@ -72,53 +72,59 @@ pub async fn get_program_swaps(
 		.map_ok(|program_swap_account_data_chunk| {
 			stream::iter(program_swap_account_data_chunk.into_iter().filter_map(
 				|(account, program_swap_account_data)| match program_swap_account_data {
-					Some(data)
-						if (data.src_token.is_none() ||
-							data.src_token.is_some_and(|addr| addr == usdc_token_mint_pubkey.into())) => {
-
-								let (deposit_metadata, vault_swap_parameters) = match data.ccm_parameters {
-									None => {
-										let CfParameters { ccm_additional_data: (), vault_swap_parameters } =
-											CfParameters::decode(&mut &data.cf_parameters[..]).map_err(|e| warn!("error while decoding CfParameters for solana vault swap: {}. Omitting swap", e)).ok()?;
-										(None, vault_swap_parameters)
+					Some(data) => {
+						let from_asset = if data.src_token.is_none() {
+							SolAsset::Sol
+						} else if data.src_token.unwrap() == usdc_token_mint_pubkey.into() {
+							SolAsset::SolUsdc
+						} else {
+							warn!("Unsupported input token for the witnessed solana vault swap, omitting the swap and the swap account.");
+							None?
+						};
```

Comment:
We can avoid the unwrap:
```rust
						let asset = if let Some(token) = data.dst_token {
							if token == usdc_token_mint_pubkey.into() {
								warn!("Unsupported output token for the witnessed solana vault swap, omitting the swap and the swap account.");
								SolAsset::SolUsdc
							} else {
								return None;
							}
						} else {
							SolAsset::Sol
						};
```

Reply:
Yeah ofc, better to use the language features so you don't even need to think about if it's safe to unwrap or not 👍